File: backend/src/Zhasyl.Api/Domain/Content/Station.cs
namespace Zhasyl.Api.Domain.Content;

public sealed class Station
{
    public Guid Id { get; set; }
    public required string Slug { get; set; }
    public ICollection<StationTranslation> Translations { get; set; } = [];
    public ICollection<Laboratory> Laboratories { get; set; } = [];
}

public sealed class StationTranslation
{
    public Guid StationId { get; set; }
    public required string Locale { get; set; }
    public required string Name { get; set; }
    public required string Location { get; set; }
    public required string Briefing { get; set; }
    public Station Station { get; set; } = null!;
}

File: frontend/src/types/station.ts
export interface MissionSummary {
  id: string;
  name: string;
  problem: string;
  status: string;
}

export interface LaboratorySummary {
  id: string;
  name: string;
  purpose: string;
  specialist: string;
  firstMission: MissionSummary;
}

export interface StationOverview {
  stationId: string;
  stationName: string;
  locale: string;
  location: string;
  briefing: string;
  laboratories: LaboratorySummary[];
}
